source/Meadow.Core/Update/UpdateService.cs

```diff
@@ -572,6 +580,49 @@ private void DisplayTree(DirectoryInfo di)
         }
     }
 
+    // In order to not require the use of a library, this was taken from
+    // RFC7515, Appendix C: https://datatracker.ietf.org/doc/html/rfc7515#appendix-C
+    private string base64UrlDecode(string arg)
+    {
+        string s = arg;
+        s = s.Replace('-', '+'); // 62nd char of encoding
+        s = s.Replace('_', '/'); // 63rd char of encoding
+        switch (s.Length % 4) // Pad with trailing '='s
+        {
+            case 0: break; // No pad chars in this case
+            case 2: s += "=="; break; // Two pad chars
+            case 3: s += "="; break; // One pad char
+            default:
+                throw new Exception("Illegal base64url string!");
```

Comment:
What's the best thing to do with exceptions like this and the exceptions that are thrown below? Should I have better error messages? Should it return null instead and handle null checking on the calling side?